File: lib/features/meeting/domain/entities/meeting.dart
import 'task.dart';

class Meeting {
  final int? id;
  final String title;
  final String? audioPath;
  final String? transcription;
  final String? summary;
  final List<String> decisions;
  final List<Task> tasks;
  final DateTime createdAt;
  final String status; // 'pending', 'processing', 'done'

  const Meeting({
    this.id,
    required this.title,
    this.audioPath,
    this.transcription,
    this.summary,
    this.decisions = const [],
    this.tasks = const [],
    required this.createdAt,
    this.status = 'pending',
  });
}

File: mindscribe_api/app/Models/Meeting.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Meeting extends Model
{
    protected $fillable = [
        'user_id', 'title', 'audio_path',
        'transcription', 'summary', 'status',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function decisions()
    {
        return $this->hasMany(Decision::class);
    }

    public function tasks()
    {
        return $this->hasMany(Task::class);
    }
}